app/scripts/filters/util.js

```diff
@@ -528,6 +528,36 @@ angular.module('openshiftConsole')
       }
       var uri = new URI(url);
       var protocol = uri.protocol();
-      return uri.is('absolute') && (protocol === 'http' || protocol === 'https');      
+      return uri.is('absolute') && (protocol === 'http' || protocol === 'https');
+    };
+  })
+  // currently used to set an alt text value for an env var that has a
+  // valueFrom attribute instead of a value.
+  .filter('altTextForValueFrom', function() {
+    return function(envVar) {
+      if(envVar.value) {
+        return;
+      }
+      if(!envVar.valueFrom) {
+        return;
+      }
+      envVar.valueIcon = 'pficon pficon-help';
+      envVar.valueIconTooltip = 'This is a referenced value that will be generated when a container is created.  On running pods you can check the resolved values by going to the Terminal tab and echoing the environment variable.';
+      _.each(envVar.valueFrom, function(from, refType) {
+        switch(refType) {
+          case 'configMapKeyRef':
+            envVar.valueAlt = 'Set to the key ' + from.key + ' in config map ' + from.name;
```

Comment:
agree with sam here, why not use what angular gives us already

On Fri, Aug 12, 2016 at 4:14 PM, Sam Padgett <notifications@github.com>
wrote:

> In app/scripts/filters/util.js
> <https://github.com/openshift/origin-web-console/pull/351#discussion_r74653754>
> :
>
> > +  // currently used to set an alt text value for an env var that has a
> > +  // valueFrom attribute instead of a value.
> > +  .filter('altTextForValueFrom', function() {
> > +    return function(envVar) {
> > +      if(envVar.value) {
> > +        return;
> > +      }
> > +      if(!envVar.valueFrom) {
> > +        return;
> > +      }
> > +      envVar.valueIcon = 'pficon pficon-help';
> > +      envVar.valueIconTooltip = 'This is a referenced value that will be generated when a container is created.  On running pods you can check the resolved values by going to the Terminal tab and echoing the environment variable.';
> > +      _.each(envVar.valueFrom, function(from, refType) {
> > +        switch(refType) {
> > +          case 'configMapKeyRef':
> > +            envVar.valueAlt = 'Set to the key ' + from.key + ' in config map ' + from.name;
>
> Curious if there's a reason you prefer this to just prefixing kve
> properties with a $. It seems like it's a bit more error prone trying to
> clean them up this way.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin-web-console/pull/351/files/26c80dd0a8c5d46c60834165a6d4e990a08f6bb7#r74653754>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABZk7cxa2tWwZMTTAsTBDvYLCRhqXMHLks5qfNQWgaJpZM4Jid4o>
> .
>